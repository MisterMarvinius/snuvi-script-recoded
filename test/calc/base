print(1 + 1);
print(1 + -3);

print(1 - 3);
print(1 - -3);

print(4 * 3);
print(6 * -3);
print(-6 * 4);
print(-6 * -4);

print(4 / 2);
print(6 / -2);
print(-6 / 3);
print(-6 / -2);
