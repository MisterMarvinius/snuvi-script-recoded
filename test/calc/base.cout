push 1
push 1
use +(2)
use print(1)
push 1
push 3
use -(1)
use +(2)
use print(1)
push 1
push 3
use -(2)
use print(1)
push 1
push 3
use -(1)
use -(2)
use print(1)
push 4
push 3
use *(2)
use print(1)
push 6
push 3
use -(1)
use *(2)
use print(1)
push 6
use -(1)
push 4
use *(2)
use print(1)
push 6
use -(1)
push 4
use -(1)
use *(2)
use print(1)
push 4
push 2
use /(2)
use print(1)
push 6
push 2
use -(1)
use /(2)
use print(1)
push 6
use -(1)
push 3
use /(2)
use print(1)
push 6
use -(1)
push 2
use -(1)
use /(2)
use print(1)
