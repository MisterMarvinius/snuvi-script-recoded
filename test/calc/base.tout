(1, LITERAL, "print")
(1, OPEN_BRACKET)
(1, NUMBER, 1.0)
(1, ADD)
(1, NUMBER, 1.0)
(1, CLOSE_BRACKET)
(1, SEMICOLON)
(2, LITERAL, "print")
(2, OPEN_BRACKET)
(2, NUMBER, 1.0)
(2, ADD)
(2, SUB)
(2, NUMBER, 3.0)
(2, CLOSE_BRACKET)
(2, SEMICOLON)
(4, LITERAL, "print")
(4, OPEN_BRACKET)
(4, NUMBER, 1.0)
(4, SUB)
(4, NUMBER, 3.0)
(4, CLOSE_BRACKET)
(4, SEMICOLON)
(5, LITERAL, "print")
(5, OPEN_BRACKET)
(5, NUMBER, 1.0)
(5, SUB)
(5, SUB)
(5, NUMBER, 3.0)
(5, CLOSE_BRACKET)
(5, SEMICOLON)
(7, LITERAL, "print")
(7, OPEN_BRACKET)
(7, NUMBER, 4.0)
(7, MUL)
(7, NUMBER, 3.0)
(7, CLOSE_BRACKET)
(7, SEMICOLON)
(8, LITERAL, "print")
(8, OPEN_BRACKET)
(8, NUMBER, 6.0)
(8, MUL)
(8, SUB)
(8, NUMBER, 3.0)
(8, CLOSE_BRACKET)
(8, SEMICOLON)
(9, LITERAL, "print")
(9, OPEN_BRACKET)
(9, SUB)
(9, NUMBER, 6.0)
(9, MUL)
(9, NUMBER, 4.0)
(9, CLOSE_BRACKET)
(9, SEMICOLON)
(10, LITERAL, "print")
(10, OPEN_BRACKET)
(10, SUB)
(10, NUMBER, 6.0)
(10, MUL)
(10, SUB)
(10, NUMBER, 4.0)
(10, CLOSE_BRACKET)
(10, SEMICOLON)
(12, LITERAL, "print")
(12, OPEN_BRACKET)
(12, NUMBER, 4.0)
(12, DIV)
(12, NUMBER, 2.0)
(12, CLOSE_BRACKET)
(12, SEMICOLON)
(13, LITERAL, "print")
(13, OPEN_BRACKET)
(13, NUMBER, 6.0)
(13, DIV)
(13, SUB)
(13, NUMBER, 2.0)
(13, CLOSE_BRACKET)
(13, SEMICOLON)
(14, LITERAL, "print")
(14, OPEN_BRACKET)
(14, SUB)
(14, NUMBER, 6.0)
(14, DIV)
(14, NUMBER, 3.0)
(14, CLOSE_BRACKET)
(14, SEMICOLON)
(15, LITERAL, "print")
(15, OPEN_BRACKET)
(15, SUB)
(15, NUMBER, 6.0)
(15, DIV)
(15, SUB)
(15, NUMBER, 2.0)
(15, CLOSE_BRACKET)
(15, SEMICOLON)
(16, EOF)
