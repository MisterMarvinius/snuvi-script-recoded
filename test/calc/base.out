2.0
-2.0
-2.0
4.0
12.0
-18.0
-24.0
24.0
2.0
-3.0
-2.0
3.0
